File: 2024/March/_16/C/exerc4_palindromo.c
#include <stdio.h>
#include <string.h>

int palindromo(char str[]);
int palindromoRec(char str[], int pos);

int palindromo(char str[])
{
    return palindromoRec(str, 0);
}

int palindromoRec(char str[], int pos)
{
    int len = strlen(str);
    if(pos == len/2)
        return 1;
    if(str[pos] != str[len-pos-1])
        return 0;
    return palindromoRec(str, pos+1);
}

int main()
{
    char s[] = "ABCCBA";
    printf("%s é palíndromo: %d\n",s,palindromo(s));
}

File: 2024/March/_16/Exerc4_Palindromo.java
package March._16;
public class Exerc4_Palindromo {

    public static boolean palindromo(String str)
    {
        return palindromoRec(str, 0);
    }

    public static boolean palindromoRec(String str, int pos)
    {
        if(pos == str.length()/2)
            return true;
        if(str.charAt(pos) != str.charAt(str.length()-pos-1))
            return false;
        return palindromoRec(str, pos+1);
    }

    public static void main(String[] args) {

        String s = "ABCCBA";
        System.out.println(s+" é palíndromo: "+palindromo(s));
    }
}
